lib/multi.c

```diff
@@ -1252,7 +1364,13 @@ static CURLMcode Curl_multi_wait(struct Curl_multi *multi,
     }
   }
 
-  return CURLM_OK;
+#ifdef USE_WINSOCK
+fail:
+  for(i = 0; i < nevents_alloced; i++)
+    WSACloseEvent(uevents[i]);
```

Comment:
Before closing the event, please make sure it is no longer assigned to a socket, like done here:
https://github.com/curl/curl/blob/f9983a6f9eb5314ad1a788d06929f6763ada6204/tests/server/sockfilt.c#L874-L875

Reply:
Are you sure this is necessary? The docs indicate that you can pass a socket with no event, but don't describe passing an event with no socket, so I'll have to iterate over the sockets again if it is.

Reply:
Yes, the docs are right and this is what I am doing in my code. It passes "wsasock" there and not "wsaevent".

Reply:
Addressed in my new version.